src/main/java/com/uid2/admin/vertx/AdminVerticle.java

```diff
@@ -25,149 +27,115 @@ public class AdminVerticle extends AbstractVerticle {
         this.services = services;
     }
 
-    @Override
     public void start(Promise<Void> startPromise) {
         final Router router = createRoutesSetup();
-
         final int portOffset = Utils.getPortOffset();
         final int port = Const.Port.ServicePortForAdmin + portOffset;
-
         vertx.createHttpServer()
-                .requestHandler(router::handle)
-                .listen(port, result -> {
-                    if (result.succeeded()) {
-                        startPromise.complete();
-                        LOGGER.info("admin verticle started");
-                    } else {
-                        startPromise.fail(result.cause());
-                    }
-                });
+                .requestHandler(router)
+                .listen(port)
+                .onSuccess(server -> {
+                    startPromise.complete();
+                    LOGGER.info("Admin verticle started on port: {}", server.actualPort());
+                })
+                .onFailure(startPromise::fail);
     }
 
     private Router createRoutesSetup() {
         final Router router = Router.router(vertx);
-
         router.route().handler(SessionHandler.create(LocalSessionStore.create(vertx)));
-        AuthHandler oauth2Handler = authHandlerFactory.createAuthHandler(vertx, router.route("/oauth2-callback"));
-
-        // protect the resource under "/adm/*"
-        router.route("/adm/*").handler(oauth2Handler);
-
-        // login page requires oauth2
-        router.route("/login").handler(oauth2Handler);
-        router.get("/login").handler(ctx -> {
-            ctx.response().setStatusCode(302)
-                    .putHeader(HttpHeaders.LOCATION, "/")
-                    .end("Redirecting to /");
-        });
-
-        // logout
-        router.get("/logout").handler(ctx -> {
-            ctx.session().destroy();
-            ctx.response().setStatusCode(302)
-                    .putHeader(HttpHeaders.LOCATION, "/")
-                    .end("Redirecting to /");
-        });
-
-        // The protected resource
-        router.get("/protected").handler(ctx -> {
-            AccessToken user = (AccessToken) ctx.user();
-            // retrieve the user profile, this is a common feature but not from the official OAuth2 spec
-            user.userInfo(res -> {
-                if (res.failed()) {
-                    ctx.session().destroy();
-                    ctx.fail(res.cause());
-                } else {
-                    final JsonObject userInfo = res.result();
-                    ctx.response().end(userInfo.getString("email"));
-                }
-            });
-        });
+
+        final OAuth2Auth oauth2Provider = (OAuth2Auth) authFactory.createAuthProvider(vertx);
+
+        final AuthenticationHandler authHandler = authFactory.createAuthHandler(vertx, router.route("/oauth2-callback"), oauth2Provider);
+        final RedirectToRootHandler redirectToRootHandler = new RedirectToRootHandler();
```

Comment:
And where has the /protected endpoint gone?

Reply:
I can add it back, is it being used anywhere? I don't see any `/protected` API calls in any of the repos

Reply:
Sure I can add it back just in case (though I'm pretty sure the `/protected` endpoint is copied from the Vert.x routing examples...)